apps/src/aichat/views/UserChatMessageEditor.tsx

```diff
@@ -30,6 +30,12 @@ const UserChatMessageEditor: React.FunctionComponent<{
 
   const disabled = isWaitingForChatResponse || saveInProgress;
 
+  useEffect(() => {
+    if (!disabled) {
+      document.getElementById('uitest-chat-textarea')?.focus();
```

Comment:
Sorry for the late review - this seems fine for now but I'm a little iffy on us relying on a UI test element ID, especially if we're referring to it by hardcoded string (and not a constant).

Using `ref` and `forwardRef` might be safer here since we can forward the ref into the message editor component and use that to focus? https://react.dev/learn/manipulating-the-dom-with-refs#accessing-another-components-dom-nodes

Reply:
Oh interesting! Took me a while to wrap my head around this but here's a follow-up PR https://github.com/code-dot-org/code-dot-org/pull/61247